chain-extensions/xvm/src/lib.rs

```diff
@@ -52,27 +51,19 @@ where
 
         match func_id {
             XvmFuncId::XvmCall => {
-                // TODO: correct weight calculation directly from pallet!
-                let weight = Weight::from_ref_time(1_000_000_000);
-                env.charge_weight(weight)?;
+                // We need to immediately charge for the worst case scenario. Gas equals Weight in pallet-contracts context.
```

Comment:
I was confused at first by this because it seems so familiar lol :sweat_smile:

Reply:
I believe gas charging should be rewritten for new weighs, including comments. Let’s do it in XVMv3

Reply:
I meant to say that the reason this code looked so familiar is that I was the author. In the PR that added weight handling. xD

But it was like 2 months ago so I've forgotten about it.

This handling should be fine - some stuff needs to be benchmarked but that won't change the current weights too much.

Reply:
Fixed